cypress/e2e/awx/resources/hosts.cy.ts

```diff
@@ -37,86 +39,156 @@ describe('host and inventory host', () => {
     cy.visit(`/infrastructure/hosts`);
     createAndEditAndDeleteHost(false, inventory);
   });
-});
 
-function createAndCheckHost(inventory_host: boolean, inventory: string) {
-  const hostName = 'E2E Inventory host ' + randomString(4);
+  it('can work with groups tab', () => {
+    const url = '/infrastructure/hosts?page=1&perPage=10&sort=name';
+    cy.createInventoryHostGroup(organization).then((result) => {
+      const { inventory, host, group } = result;
+      // TODO: unsafe assignment, find a better way to retrieve host id it doesn't return always
+      const hostid = host.id ? host.id.toString() : '';
+      navigateToHost(url, host.name, '[data-cy="name-column-cell"] a');
+      expect(host.inventory).to.eq(inventory.id);
+      expect(group.inventory).to.eq(inventory.id);
+      cy.clickLink(/^Groups$/);
+      //check edit group
+      cy.getByDataCy('edit-group').click();
+      cy.verifyPageTitle('Edit group');
+      cy.getByDataCy('name-form-group').type('-changed');
+      cy.getByDataCy('Submit').click();
+      cy.verifyPageTitle(group.name + '-changed');
+      cy.requestPost<{ name: string; inventory: number; id: number }>(
+        awxAPI`/hosts/${hostid}/groups/`,
+        {
+          name: 'E2E Group ' + randomString(5),
+          inventory: host.inventory,
+        }
+      ).then((group: { name: string; id: number }) => {
+        /// check multiple associate and disassociate
+        // disassociate
+        navigateToHost(url, host.name, '[data-cy="name-column-cell"] a');
+        cy.clickLink(/^Groups$/);
+        cy.getByDataCy('select-all').click();
+        disassociate();
+        cy.getByDataCy('empty-state-title').contains(
+          /^There are currently no groups associated with this host/
+        );
+        // Add - multi groups
+        cy.clickButton(/^Associate groups$/);
+        cy.getByDataCy('select-all').click();
+        cy.clickModalButton('Confirm');
+        cy.contains('button', 'Close').click();
+        cy.contains(group.name);
+        /// single disassociate
+        cy.searchAndDisplayResource(group.name);
```

Comment:
Update it to use another function compatible with new search component

Reply:
this isn't new - the groups search is still the same

Reply:
Leave this code like this for the moment and add a comment to wait this task to be done: https://issues.redhat.com/browse/AAP-22914